### PR TITLE
[Storage] Add support for sticky bit in DataLakeGen2 item and acl permissions 

### PR DESCRIPTION
**Pull request by yifanz0**
## Description
Add support for sticky bit in DataLakeGen2 item and acl permissions 

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from the scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** change version of module in pull request

### CODE REVIEW

File: src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1 [hunk 1]
```diff
@@ -821,7 +821,9 @@ function Test-DatalakeGen2
 		$dir1 = New-AzDataLakeGen2Item -Context $storageContext -FileSystem $filesystemName -Path $directoryPath1 -Directory -Permission rwxrwxrwx -Umask ---rwx---  -Property @{"ContentEncoding" = "UDF8"; "CacheControl" = "READ"} -Metadata  @{"tag1" = "value1"; "tag2" = "value2" }
 		Assert-AreEqual $dir1.Path $directoryPath1
         Assert-AreEqual $dir1.Permissions.ToSymbolicPermissions() "rwx---rwx"
-		$dir2 = New-AzDataLakeGen2Item -Context $storageContext -FileSystem $filesystemName -Path $directoryPath2 -Directory
+		$dir2 = New-AzDataLakeGen2Item -Context $storageContext -FileSystem $filesystemName -Path $directoryPath2 -Directory -Permission r---wx-wT -Umask --x-wx--x
+        Assert-AreEqual $dir2.Path $directoryPath2
+        Assert-AreEqual $dir2.Permissions.ToSymbolicPermissions() "r------wt"
```
Comment (BethanyZhou) [inline src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1:826]: Correct indents please

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1 (hunk 1 above)
Comment (yifanz0) [reply src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1:826]: Sorry I was having some trouble adjusting the indentation here, as the indentation in VS looks fine... 
image
Do you happen to know how to deal with this?

Comment (yifanz0) [reply src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1:826]: Also just checked this test file on main - it seems like it's misaligned everywhere while it all looks fine locally on my side :( 
https://github.com/Azure/azure-powershell/blob/main/src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1#L821

Comment (BethanyZhou) [reply src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1:826]: Yes, please use VS code and use it to convert all tabs to spaces. Then you can find the difference

File: src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs [hunk 2]
```diff
@@ -83,9 +83,10 @@ public string Source
         [ValidatePattern("([r-][w-][x-]){3}")]
         public string Umask { get; set; }
 
-        [Parameter(Mandatory = false, HelpMessage = "Sets POSIX access permissions for the file owner, the file owning group, and others. Each class may be granted read, write, or execute permission. Symbolic (rwxrw-rw-) is supported. ")]
+        [Parameter(Mandatory = false, HelpMessage = "Sets POSIX access permissions for the file owner, the file owning group, and others. Each class may be granted read, write, or execute permission. Symbolic (rwxrw-rw-) is supported. " +
+            "The sticky bit is also supported and its represented either by the letter t or T in the final character-place depending on whether the execution bit for the others category is set or unset respectively, absence of t or T indicates sticky bit not set.")]
         [ValidateNotNullOrEmpty]
-        [ValidatePattern("([r-][w-][x-]){3}")]
+        [ValidatePattern("([r-][w-][x-]){2}([r-][w-][xtT-])")]
```
Comment (blueww) [inline src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs:89]: Per [SDK](https://github.com/Azure/azure-sdk-for-net/blob/ae550cb5d4be459e61d0f42f46c00033b0fc9a83/sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathPermissions.cs#L116) , we should allow an optional "+" in the end for ExtendedAcls .
Please help to modify it to `[r-][w-][x-]){2}[r-][w-][xtT-][+]{0,1}`

File: src/Storage/Storage/DatalakeGen2/Cmdlet/UpdateAzDataLakeGen2Item.cs [hunk 3]
```diff
@@ -53,9 +53,11 @@ public class SetAzDataLakeGen2ItemCommand : StorageCloudBlobCmdletBase
...
         public AzureDataLakeGen2Item InputObject { get; set; }
 
-        [Parameter(Mandatory = false, HelpMessage = "Sets POSIX access permissions for the file owner, the file owning group, and others. Each class may be granted read, write, or execute permission. Symbolic (rwxrw-rw-) is supported. Invalid in conjunction with ACL.")]
+        [Parameter(Mandatory = false, HelpMessage = "Sets POSIX access permissions for the file owner, the file owning group, and others. Each class may be granted read, write, or execute permission. Symbolic (rwxrw-rw-) is supported." +
+            "The sticky bit is also supported and its represented either by the letter t or T in the final character-place depending on whether the execution bit for the others category is set or unset respectively, absence of t or T indicates sticky bit not set." +
+            "Invalid in conjunction with ACL.")]
         [ValidateNotNullOrEmpty]
-        [ValidatePattern("([r-][w-][x-]){3}")]
+        [ValidatePattern("([r-][w-][x-]){2}[r-][w-][xtT-]")]
         public string Permission { get; set; }
```
Comment (blueww) [inline src/Storage/Storage/DatalakeGen2/Cmdlet/UpdateAzDataLakeGen2Item.cs:61]: Per [SDK](https://github.com/Azure/azure-sdk-for-net/blob/ae550cb5d4be459e61d0f42f46c00033b0fc9a83/sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathPermissions.cs#L116) , we should allow an optional "+" in the end for ExtendedAcls .
Please help to modify it to `[r-][w-][x-]){2}[r-][w-][xtT-][+]{0,1}`

**Review by blueww — CHANGES_REQUESTED** [review]

File: src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1 (hunk 1 above)
Comment (yifanz0) [reply src/Storage/Storage.Management.Test/ScenarioTests/StorageDataPlaneTests.ps1:826]: Done. Thanks a lot for the help :)

File: src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs (hunk 2 above)
Comment (yifanz0) [reply src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs]: Done

File: src/Storage/Storage/DatalakeGen2/Cmdlet/UpdateAzDataLakeGen2Item.cs (hunk 3 above)
Comment (yifanz0) [reply src/Storage/Storage/DatalakeGen2/Cmdlet/UpdateAzDataLakeGen2Item.cs:61]: Done

File: src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs [hunk 4]
```diff
@@ -375,13 +376,20 @@ protected void SetBlobPermissionWithUMask(CloudBlockBlob blob, string permission
             string blobPermission = string.Empty;
             for (int i = 0; i < permission.Length; i++)
             {
-                if (umask[i] != '-')
+                if (Char.ToLower(permission[i]) == 't')
```
Comment (blueww) [inline src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs:379]: Also need to handle the last "+" here.

File: src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs [hunk 5]
```diff
@@ -83,9 +83,10 @@ public string Source
         [ValidatePattern("([r-][w-][x-]){3}")]
         public string Umask { get; set; }
 
-        [Parameter(Mandatory = false, HelpMessage = "Sets POSIX access permissions for the file owner, the file owning group, and others. Each class may be granted read, write, or execute permission. Symbolic (rwxrw-rw-) is supported. ")]
+        [Parameter(Mandatory = false, HelpMessage = "Sets POSIX access permissions for the file owner, the file owning group, and others. Each class may be granted read, write, or execute permission. Symbolic (rwxrw-rw-) is supported. " +
+            "The sticky bit is also supported and its represented either by the letter t or T in the final character-place depending on whether the execution bit for the others category is set or unset respectively, absence of t or T indicates sticky bit not set.")]
         [ValidateNotNullOrEmpty]
-        [ValidatePattern("([r-][w-][x-]){3}")]
+        [ValidatePattern("([r-][w-][x-]){2}[r-][w-][xtT-][+]{0,1}")]
```
Comment (blueww) [inline src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs:89]: I just test the suffix "+" looks not work on server.
And it's also not in our rest API, so we might should not add it now.
Would you please help to revert it by remover "[+]{0,1}"?

File: src/Storage/Storage/DatalakeGen2/Cmdlet/UpdateAzDataLakeGen2Item.cs (hunk 3 above)
Comment (blueww) [reply src/Storage/Storage/DatalakeGen2/Cmdlet/UpdateAzDataLakeGen2Item.cs:61]: I just test the suffix "+" looks not work on server.
And it's also not in our rest API, so we might should not add it now.
Would you please help to revert it by remover "[+]{0,1}"?

File: src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs (hunk 5 above)
Comment (yifanz0) [reply src/Storage/Storage/DatalakeGen2/Cmdlet/NewAzDataLakeGen2Item.cs]: Done

**Review by blueww — APPROVED** [review]

**Comment by blueww** [comment]
We find a .net SDK issue for this feature, and would like to hold this PR util .net SDK issue is fixed in SDK GA version.

**Pull request closed without merging**